src/Main.cpp

```diff
@@ -125,188 +75,681 @@ struct ClickControlSystem
-			else if (key.is_down(key_binding.down))
+		}
+	}
+};
+
+struct ClickButton
+    : public ecs::System
+    , public MutAccessGroupStorage<KeyBinding>
+{
+	void on_tick() override
+	{
+		const auto& keys = KeyState::get();
+		for (auto&& [entity, key_binding] : access_storage().each())
+		{
+			if(keys.is_pressed(key_binding.p))
 			{
-				sprite.change_to("pyramidplunder/archaeologist_running");
-				movement.force.y =  movement.move_force;
+				game->start_game();
+				spdlog::info("play");
+				break;
 			}
-			else
+			if(keys.is_pressed(key_binding.escape))
 			{
-				movement.force.y -= fsignf(movement.force.y)*movement.move_force;
+				//std::cout << "isExit" << std::endl;
+				//isExit = true;
+				spdlog::info("exit");
+				Engine::get_instance().quit();
 			}
+		}
+	}
 
-			if (key.is_down(key_binding.left))
+};
+
+void PushPlateBoxSystem::on_tick()
+{
+	/* For each PushPlate, check if it's colliding with a Box. */
+	for (auto&& [entity, pushplate_pos, pushplate, sprite] : MutAccessGroupStorage<Position, PushPlate, SpriteAnimation>::access_storage().each())
+	{
+		Bool collide = false;
+
+		for (auto&& [entity, box_pos, box_movement, box] : MutAccessGroupStorage<Position, Movement, Box>::access_storage().each())
+		{
+			F32 dx, dy;
+			F32 collision_width  = box_movement.collision_box_width  + 48;
+			F32 collision_height = box_movement.collision_box_height + 48;
+
+			/* Pivot has to be adjusted (to center of push plate). */
+			dx = (box_pos.xy.x+24) - pushplate_pos.xy.x;
+			dy = (box_pos.xy.y+24) - pushplate_pos.xy.y;
+
+			if (fabsf(dx) < collision_width / 2.0f && fabsf(dy) < collision_height / 2.0f)
 			{
-				replace_component<Flip>(entity, Horizontal);
-				sprite.change_to("pyramidplunder/archaeologist_running");
-				movement.force.x = -movement.move_force;
+				collide = true;
+				break;
 			}
-			else if (key.is_down(key_binding.right))
+		}
+
+		if (collide)
+		{
+			if (!pushplate.active)
 			{
-				replace_component<Flip>(entity, None);
-				sprite.change_to("pyramidplunder/archaeologist_running");
-				movement.force.x =  movement.move_force;
+				emit(PushPlateActivatedSignal{});
+				pushplate.active = true;
+				scene.pushplate_activenow++;
+				sprite.change_to("pushplate/pushplate_5");
+				spdlog::info("[o] Pushplate ACTIVE count is {} (new activated)", scene.pushplate_activenow);
+				if (scene.pushplate_activenow >= scene.pushplatecount)
+				{
+					spdlog::info("All pushplates active!");
+					game->door_open();
+					//game->level++;
+					//game->start_level(game->level);
+				}
 			}
-			else
+		}
+		else
+		{
+			if (pushplate.active)
 			{
-				movement.force.x -= fsignf(movement.force.x)*movement.move_force;
+				spdlog::info("[o] Pushplate ACTIVE count is {} (one deactivated)", scene.pushplate_activenow);
+				pushplate.active = false;
+				scene.pushplate_activenow--;
+				sprite.change_to("pushplate/pushplate_4");
 			}
-
-			// spdlog::info("force: {} {}", movement.force.x, movement.force.y);
```

Comment:
remove commented out code (in all other places as well pls, I see there are more bellow)